src/utils/memoryChecks.cpp

```diff
@@ -0,0 +1,52 @@
+/** Copyright 2017-2020 CNRS-AIST JRL and CNRS-UM LIRMM */
+
+#pragma once
+
+#include <tvm/utils/memoryChecks.h>
+
+#include <stack>
+
+namespace tvm::utils
+{
+
+std::stack<bool> malloc_is_allowed_override_;
+
+bool is_malloc_allowed_()
+{
+#ifdef EIGEN_RUNTIME_NO_MALLOC
+  return Eigen::internal::is_malloc_allowed();
+#else
+  return true;
+#endif
+}
+
+bool set_is_malloc_allowed_(bool allow)
+{
+#ifdef EIGEN_RUNTIME_NO_MALLOC
+  return Eigen::internal::set_is_malloc_allowed(allow);
+#else
+  return allow;
+#endif
+}
+
+void override_is_malloc_allowed_(bool allow)
+{
+#ifdef EIGEN_RUNTIME_NO_MALLOC
+  malloc_is_allowed_override_.push(Eigen::internal::is_malloc_allowed());
+  if(malloc_is_allowed_override_.top() != allow)
+    Eigen::internal::set_is_malloc_allowed(allow);
+#endif
+}
+
+void restore_is_malloc_allowed_()
+{
+#ifdef EIGEN_RUNTIME_NO_MALLOC
+  assert(malloc_is_allowed_override_.size() > 0
+         && "restore_is_malloc_allowed called too many times compared to override_is_malloc_allowed.");
+  if(Eigen::internal::is_malloc_allowed() != malloc_is_allowed_override_.top())
+    Eigen::internal::set_is_malloc_allowed(malloc_is_allowed_override_.top());
```

Comment:
Why do you check the current `is_malloc_allowed` state before?

Reply:
It avoids calling the function, but I guess I just mimicked the way we did similar things before.

Reply:
Internally (at least on 3.3.4 but I guess this hasn't changed a lot between Eigen versions) `is_malloc_allowed` and `set_is_malloc_allowed` call the same function so you end up with two calls (sometimes) when one call is always enough

Reply:
I removed the `if` line in 9c4703c3957f9eef036519a0c57e644a9ed5f622